solidity/contracts/isms/multisig/AbstractMerkleRootMultisigIsm.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ Internal Imports ============
+import {IInterchainSecurityModule} from "../../interfaces/IInterchainSecurityModule.sol";
+import {AbstractMultisigIsm} from "./AbstractMultisigIsm.sol";
+import {MerkleRootMultisigIsmMetadata} from "../../libs/isms/MerkleRootMultisigIsmMetadata.sol";
+import {Message} from "../../libs/Message.sol";
+import {MerkleLib} from "../../libs/Merkle.sol";
+import {CheckpointLib} from "../../libs/CheckpointLib.sol";
+
+/**
+ * @title MerkleRootMultisigIsm
+ * @notice Manages per-domain m-of-n Validator sets that are used to verify
```

Comment:
nit: This doesn't actually manage a set at all, it provides abstract logic for verifying messages given a `validatorsAndThreshold` implementation